src/v2/view-builder/internals/FormInputFactory.js

```diff
@@ -115,7 +115,8 @@ const createIdpButtons = (remediations) => {
 
     if (!_.contains(IDP.SUPPORTED_SOCIAL_IDPS, type)) {
       type = 'general-idp';
-      displayName = idpObject.idp?.name || '{ Please provide a text value }';
+      // OKTA-396684 - makes sure that custom idps always have a name
+      displayName = loc('customauth.sign.in.with.label', 'login') + ` ${idpObject.idp?.name}`;
```

Comment:
This will produce invalid translations, as we're concatenating text. Please rely on `loc`'s argument capabilities:

```suggestion
      displayName = loc('customauth.sign.in.with.label', 'login', [idpObject.idp?.name]);
```